index_search_test.go

```diff
@@ -916,3 +916,172 @@ func TestIndex_SearchWithSort(t *testing.T) {
 		})
 	}
 }
+
+func TestIndex_SearchOnNestedFileds(t *testing.T) {
+	type args struct {
+		UID                 string
+		PrimaryKey          string
+		client              *Client
+		query               string
+		request             SearchRequest
+		searchableAttribute []string
+		sortableAttribute   []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *SearchResponse
+	}{
+		{
+			name: "TestIndexBasicSearchOnNestedFields",
+			args: args{
+				UID:     "TestIndexBasicSearchOnNestedFields",
+				client:  defaultClient,
+				query:   "An awesome",
+				request: SearchRequest{},
+			},
+			want: &SearchResponse{
+				Hits: []interface{}{
+					map[string]interface{}{
+						"id": float64(5), "title": "The Hobbit",
+					},
+				},
+				NbHits:           1,
+				Offset:           0,
+				Limit:            20,
+				ExhaustiveNbHits: false,
+			},
+		},
+		{
+			name: "TestIndexBasicSearchOnNestedFieldsWithCustomClient",
```

Comment:
what is a custom client?

Reply:
We currently use the `fasthttp` library to create an `http` client but in go you have the native `net/http` library to create a client. We offer the possibility for the user to generate his own client with `net/http` or another library.